src/probnum/quad/solvers/_bayesian_quadrature.py

```diff
@@ -278,6 +269,7 @@ def integrate(
         fun: Optional[Callable],
         nodes: Optional[np.ndarray],
         fun_evals: Optional[np.ndarray],
+        rng: Union[IntLike, np.random.Generator] = np.random.default_rng(),
```

Comment:
```suggestion
        rng: Optional[np.random.Generator] = None,
```

I would suggest to make the random number generator optional. It's always better to be explicit than implicit so that when a method that actually uses random numbers is used, the user is made aware through an exception if they do not provide their own RNG instance. Otherwise the global random state might be changed without the user being aware of it. With the arbitrary backend in #581 , rng states will always have to be passed anyways, since frameworks like JAX do not have global random state.

Reply:
done